tests/common/helpers/pfc_storm.py

```diff
@@ -118,36 +130,59 @@ def _update_template_args(self):
         Populates all the vars needed by the pfc storm templates
         """
         self.extra_vars = dict()
-        self.extra_vars = { "pfc_gen_file": self.pfc_gen_file,
-                       "pfc_queue_index": self.pfc_queue_idx,
-                       "pfc_frames_number": self.pfc_frames_number,
-                       "pfc_fanout_interface": self.peer_info['pfc_fanout_interface'],
-                       "ansible_eth0_ipv4_addr": self.ip_addr,
-                       "peer_hwsku": self.peer_info['hwsku']
-                     }
+        self.extra_vars = {
+            "pfc_gen_dir": self._PFC_GEN_DIR[self.peer_device.os],
```

Comment:
remove and hardcode the path in the template else this will fail for mellanox fanouts since we haven't defined them

Reply:
This is used for `SONiC`, I didn't change the hardcode path in the template for the other systems.

Reply:
I meant the self._PFC_GEN_DIR doesn't define a path for mellanox fanouts.

Reply:
Yes, I see in LINE [84](https://github.com/Azure/sonic-mgmt/blob/91a8222cb88a1d14b3ed7e69f2222ebaedc500ea/tests/common/helpers/pfc_storm.py#L84) that it limits the fanout to only `EOS`, should I consider onyx for now?

Reply:
I get your point, thanks for reminding me!